src/crypto/crypto_openssl.c

```diff
@@ -90,6 +90,32 @@ static EC_KEY * EVP_PKEY_get0_EC_KEY(EVP_PKEY *pkey)
 
 #endif /* OpenSSL version < 1.1.0 */
 
+/**
+ * is_fips_ready - Check FIPS POST status.
+ * Returns: 0 on success, -1 on failure.
+ */
+int is_fips_ready(char *crypto_name, const size_t name_len)
+{
+	int status=-1;
+
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
+	/** Check if the OpenSSL self-tests have passed, by fetching an algorithm with
+	 *  fips=yes. This will fail if either no FIPS provider is available or
+	 *  self-tests on provider load didn't pass.
+	 */
+	EVP_CIPHER *cipher = EVP_CIPHER_fetch(NULL, "AES-256-CBC", "fips=yes");
```

Comment:
Where is this API EVP_CIPHER_fetch() implemented? 
Also not so clear on why we need a FIP_POST status check in wpa_supplicant where we don't have a real FIPS validation?